### DISCUSSION TITLE
PostCSS Import Order with base styles not working

### DISCUSSION THREAD

**Discussion by kitsunekyo** [tailwindlabs/tailwindcss#7049] [Help]
just like in issue #1057 I am adding some base style classes after `tailwind/base` and `tailwind/components`.

when I inline the styles from `button.css`, the styles work perfectly, and overwrite the tailwind styles, due to css order. (both selectors have a specificity of 010)
once I move them to the imported file, the custom styles seem to be loaded before the tailwind styles, and break.

## i can reproduce the issue with a brand new CRA5:

`yarn create react-app my-app --template=typescript`

apply these changes:

```jsx
<!-- append to src/App.tsx -->
<input
    type="submit"
    className="btn btn-primary mt-8 px-5 py-3"
    value="Login"
 />
```

```css
/* add to top of src/index.css */
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "./button.css";

@import "tailwindcss/utilities";
```

```css
/* src/button.css */
.btn {
  @apply cursor-pointer inline-block rounded-md text-base tracking-wide font-medium px-4 py-2;
}
.btn-primary {
  @apply bg-indigo-500 text-white;
}
.btn-primary:hover {
  @apply bg-indigo-400;
}
```

```js
// tailwind.config.js
module.exports = {
  content: ["./src/**/*.tsx", "./src/**/*.ts"],
  theme: {},
};
```

the applied styles in the devtools look like this:
![image](https://user-images.githubusercontent.com/8297816/149301711-90d6215c-f69e-4589-8a55-66c0239d4ba3.png)

you see that this definition overwrites our button styles, even though they *should* be loaded before our button styles.

```css
button, [type='button'], [type='reset'], [type='submit'] {
    -webkit-appearance: button;
    background-color: transparent;
    background-image: none;
}
```

**Comment by adamwathan** [comment]
You can add postcss-import with CRA5 by creating a postcss.config.js file and adding it as the first plugin 👍🏻 Same as adding Tailwind to a CRA project.

**Reply by kitsunekyo** [reply]
unfortunately thats not correct.
I've added `postcss.config.js` to my freshly created CRA5 app, with the contents outlined in the tailwind docs

https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports

```js
module.exports = {
  plugins: [
    require("postcss-import"),
    require("tailwindcss"),
    require("autoprefixer"),
  ],
};
```

you can try it with my repro steps, and see what happens to the button styles. they are overwritten by tailwind config due to import order

**Reply by kitsunekyo** [reply]
you will also see that create-react-app disables the use of custom config files in the webpack loader config:

https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/config/webpack.config.js#L143

https://webpack.js.org/loaders/postcss-loader/#config

**Comment by kitsunekyo** [comment]
I've created a PR with create-react-app to bring this feature back. 

https://github.com/facebook/create-react-app/pull/11926

**Reply by onasman** [reply]
I have the same issue with import priority order. Did you come up with a way around this before your pr is merged? @kitsunekyo

**Reply by kitsunekyo** [reply]
we could only solve it by ejecting unfortunately. with one of the latest update, CRA broke this functionality

**Comment by ragavendra** [comment]
I am having a similar issue in `NextJs` of tailwind overriding the `button, [type="button"]`. I had to re-import the `background-image` like in the below screen.

![image](https://github.com/tailwindlabs/tailwindcss/assets/5716329/98d0b556-cfc0-464d-a3ca-1125174b98da)

My `index.css` is 

```
@tailwind base;
@tailwind components;
@tailwind utilities;

.leaflet-control-geocoder-icon {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath d='M12.2 13l3.4 6.6c.6 1.1 2.5-.4 2-1.2l-4-6.2z'/%3E%3Ccircle cx='10.8' cy='8.9' r='3.9' fill='none' stroke='%23000' stroke-width='1.5'/%3E%3C/svg%3E");
}

```
